src/hooks.js

```diff
@@ -0,0 +1,179 @@
+import { getServiceName, resolvePathFromService } from './utils.js';
+import { getUrlFromRegistry } from './registries.js';
+import { getManifest, clearManifest } from './manifest.js';
+import { stepMiddleware } from './middleware.js';
+
+const config = System.sofe || {};
+
+const systemNormalize = System.normalize;
+const systemLocate = System.locate;
+
+const hasWindow = typeof window !== 'undefined';
+
+let allMiddleware = [];
+let serviceMap = {};
+let serviceNames = [];
+let middlewareMap = {};
+let middlewareId = 0;
+let middlewareTracker = 0;
+
+/**
+ * Override the default System.normalize
+ *
+ * This is done to recognize when we are loading a dependency of a sofe service and
+ * allow the dependency to load relative to the parent service.
+ */
+export function normalize(name, parentName, parentAddress) {
+	const isSofePlugin = /.+!sofe.*/;
+	name = name.match(isSofePlugin) ? normalizeSofePlugin(name) : name;
+
+	// If the module is loaded by a parent referencing !sofe, treat it is as a sofe service
+	if (parentName && parentName.match(/sofe(@[0-9a-zA-Z\-\.]+)?\.js$/)) {
+		if (name.match(isSofePlugin)) {
+			return systemNormalize.call(this, name, parentName, parentAddress);
+		} else {
+			if (name && name[0] === '.') {
+				// Only load files relative to the sofe service
+				// if they are loaded with a relative path
+				return resolvePathFromService(serviceMap, name, parentName);
+			} else {
+				// Else treat the dependency like a normal file
+				// to be loaded relative to the application code
+				return systemNormalize.call(this, name, parentName, parentAddress);
+			}
+		}
+	} else {
+		return systemNormalize.call(this, name, parentName, parentAddress);
+	}
+
+	function normalizeSofePlugin(name) {
+		return name.slice(0, name.indexOf('!')) + `!sofe`;
+	}
+}
+
+export function isOverride() {
+	return serviceNames.reduce(toIsOverriden, false);
+
+	function toIsOverriden(isOverride, serviceName) {
+		if (isOverride) return isOverride;
+		else return !!(hasWindow && (
+			window.localStorage.getItem(`sofe:${serviceName}`) ||
+			window.sessionStorage.getItem(`sofe:${serviceName}`)
```

Comment:
Should `isOverride` be based on if the service was actually loaded from the url in local storage, or just if there is a local storage value that could possibly have overridden it?

The edge case is when you set local storage after already having loaded a sofe service.